src/main/java/com/algorand/algosdk/transaction/AtomicTransactionComposer.java

```diff
@@ -0,0 +1,415 @@
+package com.algorand.algosdk.transaction;
+
+import com.algorand.algosdk.abi.Method;
+import com.algorand.algosdk.abi.ABIType;
+import com.algorand.algosdk.abi.TypeTuple;
+import com.algorand.algosdk.account.Account;
+import com.algorand.algosdk.account.LogicSigAccount;
+import com.algorand.algosdk.builder.transaction.ApplicationCallTransactionBuilder;
+import com.algorand.algosdk.crypto.Digest;
+import com.algorand.algosdk.crypto.MultisigAddress;
+import com.algorand.algosdk.util.Encoder;
+import com.algorand.algosdk.v2.client.Utils;
+import com.algorand.algosdk.v2.client.common.AlgodClient;
+import com.algorand.algosdk.v2.client.common.Response;
+import com.algorand.algosdk.v2.client.model.PendingTransactionResponse;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+
+public class AtomicTransactionComposer {
+    public enum AtomicTxComposerStatus {
+        BUILDING,
+        BUILT,
+        SIGNED,
+        SUBMITTED,
+        COMMITTED
+    }
+
+    private static final int MAX_GROUP_SIZE = 16;
+
+    private static final byte[] ABI_RET_HASH = new byte[]{0x15, 0x1f, 0x7c, 0x75};
+
+    public AtomicTxComposerStatus status;
+    public Map<Integer, Method> methodMap = new HashMap<>();
+    public List<TransactionWithSigner> transactionList = new ArrayList<>();
+    public List<SignedTransaction> signedTxns = new ArrayList<>();
+
+    public AtomicTransactionComposer() {
+        this.status = AtomicTxComposerStatus.BUILDING;
+    }
+
+    /**
+     * Get the status of this composer's transaction group.
+     */
+    public AtomicTxComposerStatus getStatus() {
+        return this.status;
+    }
+
+    /**
+     * Get the number of transactions currently in this atomic group.
+     */
+    public int getCount() {
+        return this.transactionList.size();
+    }
+
+    /**
+     * Create a new composer with the same underlying transactions. The new composer's status will be
+     * BUILDING, so additional transactions may be added to it.
+     */
+    public AtomicTransactionComposer cloneAtomicTxnComposer() throws IOException {
+        AtomicTransactionComposer cloned = new AtomicTransactionComposer();
+        cloned.signedTxns = new ArrayList<>();
```

Comment:
Good question, I am in some doubt about Java's deep copy mechanism, and I might overworked in this function. I think folks can discuss what should (or should not) be done on this thread '_>'

Reply:
No worries, I was just curious because it seems like calling the constructor will automatically initialize the following:

`public List<SignedTransaction> signedTxns = new ArrayList<>();`

Reply:
OK, now signedTxn should be automatically be an empty list.